crates/mongodb-agent-common/src/interface_types/mongo_agent_error.rs

```diff
@@ -21,13 +20,14 @@ pub enum MongoAgentError {
     MongoDBSerialization(#[from] mongodb::bson::ser::Error),
     MongoDBSupport(#[from] mongodb_support::error::Error),
     NotImplemented(&'static str),
-    MutationError(#[from] MutationError),
+    ProcedureError(#[from] ProcedureError),
```

Comment:
It should probably just be `Procedure` to be consistent with the other enum values
```suggestion
    Procedure(#[from] ProcedureError),
```